types/src/v17/util/error.rs

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: CC0-1.0
+
+use core::fmt;
+
+use bitcoin::blockdata::script::{witness_program, witness_version};
+use bitcoin::{address, amount, hex};
+
+/// Error when converting a `CreateMultisig` type into the model type.
+#[derive(Debug)]
+pub enum CreateMultisigError {
+    /// Conversion of the `address` field failed.
+    Address(address::ParseError),
+    /// Conversion of the `redeem_script` field failed.
+    RedeemScript(hex::HexToBytesError),
+}
+
+impl fmt::Display for CreateMultisigError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use CreateMultisigError as E;
+
+        match *self {
+            E::Address(ref e) => write!(f, "conversion of the `address` field failed: {}", e),
+            E::RedeemScript(ref e) =>
+                write!(f, "conversion of the `redeem_script` field failed: {}", e),
+        }
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for CreateMultisigError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        use CreateMultisigError as E;
+
+        match *self {
+            E::Address(ref e) => Some(e),
+            E::RedeemScript(ref e) => Some(e),
+        }
+    }
+}
+
+/// Error when converting an `EstimateSmartFee` type into the model type.
+#[derive(Debug)]
+pub enum EstimateSmartFeeError {
+    /// Conversion of the `smart_fee` field failed.
+    SmartFee(amount::ParseAmountError),
```

Comment:
While  not incorrect, since there is only one error I'd just return the `ParseAmountError` directly. Like you do with `SignMessageWithPrivKey`.